syntax = "proto3";

package sila2;

// Basic SiLA2 Device Service
service SiLA2Device {
  rpc GetDeviceInfo(DeviceInfoRequest) returns (DeviceInfoResponse);
  rpc ExecuteCommand(CommandRequest) returns (CommandResponse);
  rpc ListDevices(ListDevicesRequest) returns (ListDevicesResponse);
}

// Device Info Messages
message DeviceInfoRequest {
  string device_id = 1;
}

message DeviceInfoResponse {
  string device_id = 1;
  string status = 2;
  string device_type = 3;
  map<string, string> properties = 4;
  string timestamp = 5;
}

// Command Messages
message CommandRequest {
  string device_id = 1;
  string operation = 2;
  map<string, string> parameters = 3;
}

message CommandResponse {
  string device_id = 1;
  string operation = 2;
  bool success = 3;
  string status = 4;
  map<string, string> result = 5;
  string timestamp = 6;
}

// List Devices Messages
message ListDevicesRequest {
  // Empty for now
}

message ListDevicesResponse {
  repeated DeviceInfo devices = 1;
  int32 count = 2;
  string timestamp = 3;
}

message DeviceInfo {
  string device_id = 1;
  string device_type = 2;
  string status = 3;
  string location = 4;
}